tool/release/createnotes/CreateNotes.kt

```diff
@@ -21,27 +21,36 @@
 
 package com.vaticle.dependencies.tool.release.createnotes
 
+import java.nio.file.Files
+import java.nio.file.Path
 import java.nio.file.Paths
 import kotlin.io.path.notExists
 
 fun main(args: Array<String>) {
     val bazelWorkspaceDir = Paths.get(getEnv("BUILD_WORKSPACE_DIRECTORY"))
     val githubToken = getEnv("CREATE_NOTES_TOKEN")
-    if (args.size != 5) throw RuntimeException("org, repo, version, commit, and template must be supplied")
-    val (org, repo, version, commit, templateFileLocation) = args
+    if (args.size != 5) throw RuntimeException("org, repo, commit, version, and template must be supplied")
+    val (org, repo, commit, version, templateFileLocation) = args
     val templateFile = bazelWorkspaceDir.resolve(templateFileLocation)
     if (templateFile.notExists()) throw RuntimeException("Template file '$templateFile' does not exist.")
 
-    println("Repository: $org/$repo")
-    println("Commit: $commit")
+    println("Commit: $org/$repo@$commit")
     println("Version: $version")
 
-    val commits = getCommits(org, repo, Version.parse(version), commit, bazelWorkspaceDir, githubToken)
-    println("There are ${commits.size} commits to be collected.")
-    val commitDescriptions = getCommitDescriptions(org, repo, commits, githubToken)
-    writeReleaseNoteMd(commitDescriptions, templateFile)
+    val commits = collectCommits(org, repo, commit, Version.parse(version), bazelWorkspaceDir, githubToken)
+    println("Found ${commits.size} commits to be collected into the release note.")
+    val notes = collectNotes(org, repo, commits, githubToken)
+    writeNotesMd(notes, templateFile)
 }
 
 private fun getEnv(env: String): String {
     return System.getenv(env) ?: throw RuntimeException("'$env' environment variable must be set.")
-}
\ No newline at end of file
+}
+
+private fun writeNotesMd(notes: List<Note>, releaseTemplateFile: Path) {
+    val template = String(Files.readAllBytes(releaseTemplateFile), Charsets.UTF_8)
+    if (!template.matches("[\\s\\S]*${Constant.releaseTemplateRegex.pattern}[\\s\\S]*".toRegex()))
```

Comment:
@lolski `[\\s\\S]` is equivalent to "any character" isn't it?

```suggestion
    if (!template.matches(".*${Constant.releaseTemplateRegex.pattern}.*".toRegex()))
```

Reply:
Yes, specifically "any character, including line breaks".

The `.` pattern is my first choice, but apparently that matches "any character except for line breaks".

Reply:
@lolski 

>The regular expression . matches any character except a line terminator unless the DOTALL flag is specified.

Can we use `.` + `DOTALL` as a regex flag instead of `[\\s\\S]`?

Reply:
Fixed in c1b68e07d5d1f263c627c821fe243c337ac2766f